All-in-one so far :-)

Crazy-Hopper opened the pull request:
Hi could you please take a look at this fork and apply patches you like?

Crazy-Hopper committed:
Hush the NetWinInfo deprecation warning.

Crazy-Hopper committed:
Hush the int narrowing warning.

Crazy-Hopper committed:
Create the container window on-screen.

That aids Wine icons to be drawn immediately.

Crazy-Hopper committed:
Tiny bit of code cosmetics.

Crazy-Hopper committed:
Switch to 32x32 pixel icons and avoid scaling.

Crazy-Hopper committed:
Made sure container window is under mouse upon click.

davidedmundson commented on sniproxy.cpp:
```diff
@@ -311,18 +305,38 @@ void SNIProxy::sendClick(uint8_t mouseButton, int x, int y)
 
-    //set our window so the middle is where the mouse is
+    auto cookie = xcb_query_pointer(c, m_windowId);
+    QScopedPointer<xcb_query_pointer_reply_t> pointer(xcb_query_pointer_reply(c, cookie, Q_NULLPTR));
```
Oh that's a good idea. 

Potentially mean we get a menu in a weird place if the SNI is activated via keyboard, but this solves a few problems like if your systray icons are larger than 32 and you click the bottom right

I was going to just patch Plasma..but this works too

Crazy-Hopper replied:
Oh.... problem!
xchat icon disappears completely from tray upon proxy restart.
Not even an empty space.

Crazy-Hopper replied:
That happens to your master HEAD too, so that deserves another issue report. :-)

Crazy-Hopper replied:
> Potentially mean we get a menu in a weird place if the SNI is activated via keyboard

I don't know how to activate SNI with keyboard... But won't that pass x*y of plasma space?

davidedmundson replied:
it will, but the mouse will always be out of the geometry of x and y.

For keyboard activation:

Old version: menu won't always appear, but if it does it'll be in the right place
your version: menu should always appear, but will always be in wrong place

Lets go with yours

Crazy-Hopper replied:
How do I activate an SNI with keyboard?

davidedmundson replied:
No idea, my job of code reviews is to spot hypothetical problems before they turn into real ones.

There should be better keyboard navigation in the system tray than there is... another job for future Dave.

davidedmundson commented on sniproxy.cpp:
```diff
@@ -311,18 +305,38 @@ void SNIProxy::sendClick(uint8_t mouseButton, int x, int y)
+	configVals[1] = pointer->root_y;
+    } else {
+	if (pointer->root_x > x + s_embedSize)
+	    configVals[0] = pointer->root_x - s_embedSize + 1;
```
I don't understand. 

This positions the window so we're always nearly clicking the bottom right?

davidedmundson commented on sniproxy.cpp:
```diff
@@ -157,23 +157,17 @@ SNIProxy::SNIProxy(xcb_window_t wid, QObject* parent):
-    //this is needed as chormium and such when resized just fill the icon with transparent space and only draw in the middle
-    //however spotify does need this as by default the window size is 900px wide.
-    //use an artbitrary heuristic to make sure icons are always sensible
-    if (clientGeom->width < 12 || clientGeom->width > s_embedSize ||
```
I don't like this change, they get all different sizes.

I appreciate it solves the scaling problem, but then we're just in "which problem is more annoying" which is personal preference.

Could you mabye try this patch on plasma-frameworks: https://paste.kde.org/p1adgfb1h
it might make the upscaling better; it scales in QPixmap space rather than in OpenGL on the fly 

(though both should be bilinear scaling...so probably not much different)

Crazy-Hopper replied:
Yes, the sizes are different but those sizes are original ones...
Ok, I'll try that smooth scaling on pixmap. Hope that would actually make a difference.

Crazy-Hopper replied:
Wow. A huge one! Could you please explain why you are resizing for cases <12 instead of scaling?
I'd resize to s_embedSize everything that's larger and smoothly scale everything that's smaller.

Crazy-Hopper replied:
This approach does the good job for TeamViewer (20) and Dropbox (24), but not so well for Icedove (24 too).

Crazy-Hopper replied:
Never mind, intermittent glitch.

Crazy-Hopper committed:
Revert "Fix transparent icons in runtime."

This reverts commit f1890f50d723301ab6be0ed9ac7f8d14f5482594.
It didn't solve anything for me but broke Psi icon animation.

Crazy-Hopper committed:
Reiterate "Switch to 32x32 pixel icons and avoid scaling."

This reverts commit 152e867646f5dd16cf078455315e9f52384cf2c7.
Actually, scaling is not that bad if done right. :-)
But still 32 seems to be a must.

Crazy-Hopper committed:
Reiterate "Fix transparent icons in runtime."

This brings back in the commit f1890f50d723301ab6be0ed9ac7f8d14f5482594.

Even though this commit breaks Psi (and probably others with analogous
icon animation) having empty icons is way worse.

Crazy-Hopper committed:
Merge remote-tracking branch 'upstream/master'

Crazy-Hopper committed:
Emit NewToolTip on icon update.

This helps native KDE SNI tooltip to be updated with the new icon.

Crazy-Hopper committed:
Shorten up the transparency checking runtime.

Mostly by checking center and sub-center pixels.
And by fixing the breakage of the loop.

Crazy-Hopper commented:
Not a single commit? :-( Oh well.
